src/visualization/context/pagination.tsx

```diff
@@ -59,7 +59,7 @@ export const PaginationProvider: FC<PaginationProviderProps> = ({
     } else {
       setOffset(offset + size)
     }
-  }, [offset, size, setOffset]) // eslint-disable-line react-hooks/exhaustive-deps
+  }, [offset, size, setOffset, total])
```

Comment:
The linter is correct here. Instead of disabling the rule, we should satisfy the linter and the hook will work correctly.

Reply:
This is like, the perfect example of what we were talking about @ code review yesterday